src/deepsparse/server/server.py

```diff
@@ -237,16 +249,76 @@ def _add_endpoint(
     pipeline = Pipeline.from_config(pipeline_config, context, server_logger)
 
     _LOGGER.info(f"Adding endpoints for '{endpoint_config.name}'")
-    _add_pipeline_endpoint(
+    _add_inference_endpoints(
         app,
         endpoint_config,
         server_config.system_logging,
         pipeline,
         server_config.integration,
     )
+    _add_status_and_metadata_endpoints(
+        app, endpoint_config, pipeline, server_config.integration
+    )
+
+
+def _add_endpoint_to_app(app, routes_and_fns, response_model, methods, tags):
+    for route, endpoint_fn in routes_and_fns:
+        app.add_api_route(
+            route,
+            endpoint_fn,
+            response_model=response_model,
+            methods=methods,
+            tags=tags,
+        )
+        _LOGGER.info(f"Added '{route}' endpoint")
+
+
+def clean_up_route(route):
+    if not route.startswith("/"):
+        route = "/" + route
+    return route
+
+
+def _add_status_and_metadata_endpoints(
+    app: FastAPI,
+    endpoint_config: EndpointConfig,
+    pipeline: Pipeline,
+    integration: str = INTEGRATION_LOCAL,
+):
+    def _pipeline_ready():
+        return CheckReady(status="OK")
+
+    def _model_metadata():
+        if not pipeline or not pipeline.model_path:
+            HTTPException(status_code=404, detail="Model path not found")
+        return ModelMetaData(model_path=pipeline.model_path)
+
+    routes_and_fns = []
+    meta_and_fns = []
+
+    if integration == INTEGRATION_LOCAL:
+        if endpoint_config.route:
+            endpoint_config.route = clean_up_route(endpoint_config.route)
+            route_ready = f"{endpoint_config.route}/ready"
+            route_meta = endpoint_config.route
+        else:
+            route_ready = f"/v2/models/{endpoint_config.name}/ready"
+            route_meta = f"/v2/models/{endpoint_config.name}"
+
+    elif integration == INTEGRATION_SAGEMAKER:
+        route_ready = "/invocations/ready"
+        route_meta = "/invocations"
```

Comment:
This might be overcomplicating things, but I'm thinking it could make sense here to break this out into a class, especially if we plan on supporting more integrations in the future. Ie a base route class and then subclasses for each integration.

Reply:
Yah I agree. Was planning on doing something similar as part of OpenAI integration. This is just updating routes for existing integrations